src/probnum/random_variables/_categorical.py

```diff
@@ -0,0 +1,105 @@
+"""Categorical random variables."""
+from typing import Optional
+
+import numpy as np
+
+from probnum.type import RandomStateArgType
+
+from ._random_variable import DiscreteRandomVariable
+
+
+class Categorical(DiscreteRandomVariable):
+    """Categorical random variable.
+
+    Parameters
+    ----------
+    probabilities :
+        Probabilities of the events.
+    support :
+        Support of the categorical distribution. Optional. Default is None,
+        in which case the support is chosen as :math:`(0, ..., K-1)` where
+        :math:`K` is the number of elements in `event_probabilities`.
+    random_state :
+        Random state of the random variable.
+    """
+
+    def __init__(
+        self,
+        probabilities: np.ndarray,
+        support: Optional[np.ndarray] = None,
+        random_state: Optional[RandomStateArgType] = None,
+    ):
+        # The set of events is names "support" to be aligned with the method
+        # DiscreteRandomVariable.in_support().
+
+        num_categories = len(probabilities)
+        self._probabilities = np.asarray(probabilities)
```

Comment:
We should probably enforce the sum-to-unity constraint for the probabilities, right?
For sampling, `np.random.choice` will throw an error, but `_pmf_categorical` will not complain returning unreasonable values.

Reply:
I am not sure. In principle, I would like to let this implementation not fail prematurely, if that makes sense. Who knows which kind of duck typing one would need in the future :) However, I see the point that perhaps we would like to throw meaningful errors. For the time being, I think that `np.random.choice` would return a sufficiently meaningful error message, but if you want me to change that, I will do so!

Reply:
My only concern is the `pmf` function. If you have something in mind for which the check would be troublesome, then I'll leave that decision to you :)